### PR TITLE
ZIF_ABAPGIT_LOG: extend and move UI methods to new class

### PR DESCRIPTION
**Pull request by mfsap**
Enhance ZIF_ABAPGIT_LOG
- add IS_ITEM structure (object) to message adding functions
- add method "add_exception"
- add method "add_info"
- relocate methods "show", "to_html" and "write" to new class ZCL_ABAPGIT_LOG_VIEWER

Introduction of new log viewer class ZCL_ABAPGIT_LOG_VIEWER

Replacement of relocated methods where used

-> This is the next step providing a log for serialzation/de-serialization (see #2534)

### CODE REVIEW

**Comment by mfsap** [comment]
Hmm, I created a draft PR before creating the real PR. Unfortunately the usual automatic check procedures are not executed (e.g. abaplint). How can I trigger this manually at the current PR state?

**Comment by mfsap** [comment]
All checks passed - it seems that it was just a delay in automatic execution :-)

**Comment by larshp** [comment]
syntax checked ok

File: src/utils/zif_abapgit_log.intf.abap [hunk 1]
```diff
@@ -1,21 +1,62 @@
 INTERFACE zif_abapgit_log
   PUBLIC .
 
+  TYPES:
+    BEGIN OF ty_log_out,
```
Comment (larshp) [inline src/utils/zif_abapgit_log.intf.abap]: can some of these public types be changed to private?

Comment (mfsap) [reply src/utils/zif_abapgit_log.intf.abap]: Interfaces may not have private attributes or methods. And the types are used as output of the `ZCL_ABAPGIT_LOG` and handled by `ZCL_ABAPGIT_LOG_VIEWER`.

Comment (larshp) [reply src/utils/zif_abapgit_log.intf.abap]: yea, but something seems strange, will move this to separate issue and merge this PR

**Pull request merged**